lib/Dialect/ESI/ESIStdServices.cpp

```diff
@@ -21,18 +21,24 @@
 using namespace circt;
 using namespace circt::esi;
 
-/// Wrap types in esi channels and return the port info struct.
-static ServicePortInfo createPort(StringRef name, Type toServerInner,
-                                  Type toClientInner) {
-  assert(toServerInner || toClientInner);
-  auto *ctxt =
-      toServerInner ? toServerInner.getContext() : toClientInner.getContext();
-  return {StringAttr::get(ctxt, name), ChannelType::get(ctxt, toServerInner),
-          ChannelType::get(ctxt, toClientInner)};
+/// Utility function to create a req/resp pair bundle service port.
+static ServicePortInfo createReqResp(StringAttr sym, Twine name,
+                                     StringRef reqName, Type reqType,
+                                     StringRef respName, Type respType) {
+  assert(reqType || respType);
+  auto *ctxt = reqType ? reqType.getContext() : respType.getContext();
+  auto bundle = ChannelBundleType::get(
+      ctxt,
+      {BundledChannel{StringAttr::get(ctxt, reqName), ChannelDirection::to,
+                      ChannelType::get(ctxt, reqType)},
+       BundledChannel{StringAttr::get(ctxt, respName), ChannelDirection::from,
+                      ChannelType::get(ctxt, respType)}},
```

Comment:
As i read it, it's acceptable behaviour that (xor) one of these channels may have a null type used in the `ChannelType`?

Reply:
Nope. There's a typo in the assert.